### PR TITLE
TPT 1989: Migrate `nb` datasource

### PR DESCRIPTION
**Pull request by yec-akamai**
## 📝 Description

Migrate the nodebalancer datasource from SDKv2 to terraform framework, and remove the SDKv2 implementations.

## ✔️ How to Test

`make PKG_NAME="linode/nb" testacc`

### CODE REVIEW

**Comment by yec-akamai** [comment]
/acctest sha=c02bbba294861ead82052be59f3f673f2dff218a pkg=linode/nb

File: linode/nb/framework_models.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+func parseTransfer(
+	ctx context.Context,
+	transfer linodego.NodeBalancerTransfer,
+) (*basetypes.ListValue, diag.Diagnostics) {
+	result := make(map[string]attr.Value)
+
+	result["in"] = helper.Float64PointerValueWithDefault(transfer.In)
+	result["out"] = helper.Float64PointerValueWithDefault(transfer.Out)
+	result["total"] = helper.Float64PointerValueWithDefault(transfer.Total)
```
Comment (lgarber-akamai) [inline linode/nb/framework_models.go:51]: Nice! I just validated that these helper functions mimic the logic of the SDKv2 implementation 👍

**Review by lgarber-akamai — APPROVED** [review]
Great work!

**Review by ezilber-akamai — APPROVED** [review]
Nice work!

**Comment by ezilber-akamai** [comment]
Can you change the target branch from `dev` to `framework-migration`phase-1`

**Pull request merged**